osf/models/osf_group.py

```diff
@@ -0,0 +1,557 @@
+import logging
+import functools
+from django.apps import apps
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from guardian.shortcuts import assign_perm, remove_perm, get_perms, get_objects_for_group, get_group_perms
+from guardian.models import GroupObjectPermissionBase, UserObjectPermissionBase
+
+from addons.base.utils import disconnect_addons
+from framework.exceptions import PermissionsError
+from framework.auth.core import get_user, Auth
+from framework.sentry import log_exception
+from osf.models import base
+from osf.models.mixins import GuardianMixin, Loggable
+from osf.models import AbstractNode, OSFUser, NodeLog
+from osf.models.osf_grouplog import OSFGroupLog
+from osf.utils.permissions import ADMIN, MANAGER, MEMBER, MANAGE, reduce_permissions
+from osf.utils import sanitize
+from website.project import signals as project_signals
+from website.osf_groups import signals as group_signals
+from website.util import api_v2_url
+
+logger = logging.getLogger(__name__)
+
+
+class OSFGroup(GuardianMixin, Loggable, base.ObjectIDMixin, base.BaseModel):
+    """
+    OSFGroup model.  When an OSFGroup is created, a manager and member Django group are created.
+    Managers belong to both manager and member groups.  Members belong to the member group only.
+
+    The OSFGroup's Django member group is given permissions to nodes, so all OSFGroup members
+    get the same permission to the node.
+    """
+
+    name = models.TextField(blank=False)
+    creator = models.ForeignKey(OSFUser,
```

Comment:
Similar to creator on the Node model, this field has null=True, so if the creator is deleted, this can be set to null.  However, an OSFGroup needs a creator.